src/pages/_app.js

```diff
@@ -34,6 +40,31 @@ function App({ Component, pageProps }) {
     }
   }, []);
 
+  useEffect(() => {
+    // This effect is called everytime the user changes because firebase has updated the token
+    // or if the router changes (i.e the user navigates)
+    const checkTermsAcceptance = async (u) => {
+      const serverUser = await axiosGet('/profile',
+        {
+          headers: {
+            authorization: `Bearer ${u.xa}`,
+          },
+        });
+      return serverUser.termsAccepted;
+    };
+    if (user && user.emailVerified) {
+      checkTermsAcceptance(user).then((termsAccepted) => {
```

Comment:
If I'm reading this correctly… sounds like the `GET /profile` API call will be made on every page transition? Can we set a local state the first time we detect that terms have already been accepted so that we don't have to check again (at least for the time the app is in memory / open in the browser tab).

Reply:
Removed the check on the router, so it will only happen if the user actually changes.

Reply:
It's tricky and not recommended to connect to the store in this file